### ISSUE TITLE
Uncaught TypeError: undefined is not a function

### ISSUE THREAD

**Issue by lorenzoambrosi** [atom-minimap/minimap#288]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 0.175.0
**System**: linux 3.13.0-46-generic
**Thrown From**: [minimap](https://github.com/atom-minimap/minimap) package, v4.5.0

### Stack Trace

Uncaught TypeError: undefined is not a function

```
At /home/thunder/.atom/packages/minimap/lib/minimap-element.coffee:69

TypeError: undefined is not a function
  at atom-text-editor-minimap.MinimapElement.attachedCallback (/home/thunder/.atom/packages/minimap/lib/minimap-element.coffee:69:35)
  at atom-text-editor-minimap.MinimapElement.attachToRight (/home/thunder/.atom/packages/minimap/lib/minimap-element.coffee:98:33)
  at atom-text-editor-minimap.MinimapElement.swapMinimapPosition (/home/thunder/.atom/packages/minimap/lib/minimap-element.coffee:104:8)
  at observeConfig.minimap.displayMinimapOnLeft (/home/thunder/.atom/packages/minimap/lib/minimap-element.coffee:40:10)
  at /usr/share/atom/resources/app/src/config.js:623:16
  at /usr/share/atom/resources/app/src/config.js:640:20
  at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app/node_modules/event-kit/lib/emitter.js:82:11)
  at Config.module.exports.Config.emitChangeEvent (/usr/share/atom/resources/app/src/config.js:751:29)
  at Config.module.exports.Config.setRawValue (/usr/share/atom/resources/app/src/config.js:617:19)
  at Config.module.exports.Config.set (/usr/share/atom/resources/app/src/config.js:231:14)
  at Object.subscriptions.add.subscribeTo.mousedown [as __NONE__] (/home/thunder/.atom/packages/minimap/lib/minimap-quick-settings-element.coffee:67:21)
  at HTMLButtonElement.listener (/home/thunder/.atom/packages/minimap/node_modules/atom-utils/lib/mixins/events-delegation.js:108:41)

```

### Commands

```
     -2:59.0 find-and-replace:select-next (atom-text-editor.editor)
     -2:57.2 core:save (atom-text-editor.editor)
  5x -2:27.7 wheel (atom-text-editor.editor)
     -2:07.4 core:confirm (atom-text-editor.editor.mini)
 54x -1:09.7 wheel (span.package-description)
     -0:00.0 core:confirm (atom-text-editor.editor.mini)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "file-type-icons"
    ],
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ]
  },
  "minimap": {}
}
```

### Installed Packages

```coffee
# User
file-icons, v1.5.0
minimap, v4.5.0
tree-view-breadcrumb, v0.4.1
tree-view-open-files, v0.2.3

# Dev
No dev packages
```

/cc @atom/core

**Comment by abe33** [comment]
I think you got caught in this issue atom/settings-view#377 with the settings view. 
You shouldn't have been able to install minimap v4.5.0 on Atom < v0.184. 

Can you update Atom? If not you can try downgrading the minimap package by uninstalling it and then reinstalling it with `apm install minimap@4.3.1` (4.3.1 should be the latest version compatible with 0.175).

**Comment by sbromberger** [comment]
Hi - this is still happening with Atom 0.206.0 and minimap 4.9.3:

```
TypeError: undefined is not a function
    at Decoration.module.exports.Decoration.setProperties (/Applications/Atom.app/Contents/Resources/app.asar/src/decoration.js:135:28)
    at new Decoration (/Applications/Atom.app/Contents/Resources/app.asar/src/decoration.js:58:12)
    at Minimap.module.exports.DecorationManagement.decorateMarker (/Users/seth/.atom/packages/minimap/lib/mixins/decoration-management.coffee:159:20)
    at MinimapGitDiffBinding.module.exports.MinimapGitDiffBinding.markRange (/Users/seth/.atom/packages/minimap-git-diff/lib/minimap-git-diff-binding.coffee:94:20)
    at MinimapGitDiffBinding.module.exports.MinimapGitDiffBinding.addDecorations (/Users/seth/.atom/packages/minimap-git-diff/lib/minimap-git-diff-binding.coffee:63:30)
    at MinimapGitDiffBinding.module.exports.MinimapGitDiffBinding.updateDiffs (/Users/seth/.atom/packages/minimap-git-diff/lib/minimap-git-diff-binding.coffee:51:21)
    at /Users/seth/.atom/packages/minimap-git-diff/lib/minimap-git-diff-binding.coffee:3:61
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
    at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1207:25
```

**Comment by plttn** [comment]
Disabling minimap-selection seems to stop it from happening.

**Comment by sbromberger** [comment]
@plttn I don't have minimap-selection installed. I have minimap and minimap-git-diff.

**Comment by plttn** [comment]
@sbromberger Weird, that was causing it for me.

**Comment by abe33** [comment]
It's not the same stack trace so its not the same issue, it's the same as #335 though.
Anyway, it was fixed in 4.9.4.
